vm/src/stdlib/ast.rs

```diff
@@ -502,80 +507,82 @@ fn expression_to_ast(vm: &VirtualMachine, expression: &ast::Expression) -> PyObj
                 Some(value) => expression_to_ast(vm, value),
                 None => vm.ctx.none(),
             };
-            vm.ctx.set_attr(&node, "value", py_value);
+            vm.set_attr(&node, "value", py_value).unwrap();
 
             node
         }
         ast::Expression::YieldFrom { value } => {
             let node = create_node(vm, "YieldFrom");
 
             let py_value = expression_to_ast(vm, value);
-            vm.ctx.set_attr(&node, "value", py_value);
+            vm.set_attr(&node, "value", py_value).unwrap();
 
             node
         }
         ast::Expression::Subscript { a, b } => {
             let node = create_node(vm, "Subscript");
 
             let py_value = expression_to_ast(vm, a);
-            vm.ctx.set_attr(&node, "value", py_value);
+            vm.set_attr(&node, "value", py_value).unwrap();
 
             let py_slice = expression_to_ast(vm, b);
-            vm.ctx.set_attr(&node, "slice", py_slice);
+            vm.set_attr(&node, "slice", py_slice).unwrap();
 
             node
         }
         ast::Expression::Attribute { value, name } => {
             let node = create_node(vm, "Attribute");
 
             let py_value = expression_to_ast(vm, value);
-            vm.ctx.set_attr(&node, "value", py_value);
+            vm.set_attr(&node, "value", py_value).unwrap();
 
             let py_attr = vm.ctx.new_str(name.to_string());
-            vm.ctx.set_attr(&node, "attr", py_attr);
+            vm.set_attr(&node, "attr", py_attr).unwrap();
```

Comment:
Instead of unwrap, we should you the `?` operator and return a result instead of a pythonRef.

Reply:
I can't because that function isn't PyResult.

The whole ast module needs a bit of a refactor. Every time I change something I spend most of my time there. I'll do that later though.

Reply:
Okay, good idea.